src/hotspot/share/runtime/os.cpp

```diff
@@ -1528,6 +1528,53 @@ bool os::set_boot_path(char fileSep, char pathSep) {
   return false;
 }
 
+static char* _image_release_file_content = nullptr;
+
+void os::read_image_release_file() {
+  const char* home = Arguments::get_java_home();
+  stringStream ss;
+  ss.print("%s/release", home);
+
+  if (_image_release_file_content == nullptr) {
```

Comment:
Shouldn't we check before doing anything else i.e. make line 1538 first.

Also for race concerns use load_acquire on `image_release_file_content`.

Reply:
Though actually shouldn't this just be an assert as we only expect this to be called once.